From 707a99fe27b94f48f2f4ce38c2464ba3b29c1e60 Mon Sep 17 00:00:00 2001
From: Rico Tzschichholz <ricotz@ubuntu.com>
Date: Fri, 20 Mar 2020 07:51:53 +0100
Subject: [PATCH] Fix build with boost < 1.68

Follow-up fix for 73cac1031131021819a0fbd4d60554196aea230c
"tdf130839: Corrects second level left margin in Smartart"

Change-Id: Ie194062ba33cf505ad33fbfdc73c2f764b61d0aa
---

diff --git a/oox/source/drawingml/diagram/diagramlayoutatoms.cxx b/oox/source/drawingml/diagram/diagramlayoutatoms.cxx
index 1b6a821..ff83dde 100644
--- a/oox/source/drawingml/diagram/diagramlayoutatoms.cxx
+++ b/oox/source/drawingml/diagram/diagramlayoutatoms.cxx
@@ -1224,13 +1224,13 @@
                 aParagraph->getProperties().setLevel(nLevel);
                 if (nLevel >= nStartBulletsAtLevel)
                 {
-                    if (!aParagraph->getProperties().getParaLeftMargin().has_value())
+                    if (aParagraph->getProperties().getParaLeftMargin().get_ptr() == nullptr)
                     {
                         sal_Int32 nLeftMargin = 285750 * (nLevel - nStartBulletsAtLevel + 1) / EMU_PER_HMM;
                         aParagraph->getProperties().getParaLeftMargin() = nLeftMargin;
                     }
 
-                    if (!aParagraph->getProperties().getFirstLineIndentation().has_value())
+                    if (aParagraph->getProperties().getFirstLineIndentation().get_ptr() == nullptr)
                         aParagraph->getProperties().getFirstLineIndentation() = -285750 / EMU_PER_HMM;
 
                     // It is not possible to change the bullet style for text.
